src/main/scala/scala/tools/partest/nest/AbstractRunner.scala

```diff
@@ -168,18 +167,24 @@ abstract class AbstractRunner {
       val expectedFailureMessage = if (expectedFailures == 0) "" else s" (expecting $expectedFailures to fail)"
       echo(s"Selected $totalTests tests drawn from $testContributors$expectedFailureMessage\n")
 
+      var limping = false
       val (_, millis) = timed {
         for ((kind, paths) <- grouped) {
           val num = paths.size
           val ss = if (num == 1) "" else "s"
           comment(s"starting $num test$ss in $kind")
-          val results = suiteRunner.runTestsForFiles(paths map (_.jfile.getAbsoluteFile), kind)
-          val (passed, failed) = results partition (_.isOk)
-
-          passedTests ++= passed
-          failedTests ++= failed
-          if (failed.nonEmpty) {
-            comment(passFailString(passed.size, failed.size, 0) + " in " + kind)
+          if (limping) comment(s"suite already in failure, skipping") else {
+            val results = suiteRunner.runTestsForFiles(paths.map(_.jfile.getAbsoluteFile)) match {
+              case Left((e, states)) => limping = true ; states
```

Comment:
Shouldn't the unused `e` warn? maybe that was one of the todo's noted in the tests for warnings.